pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp

```diff
@@ -1077,6 +1139,26 @@ LogicalResult ConvertAtenOp<AtenSizeIntOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenNumelOp>::matchAndRewrite(
+    AtenNumelOp op,
+    OpAdaptor adaptor,
+    ConversionPatternRewriter& rewriter) const {
+  auto outType = op.getType().dyn_cast<TensorType>();
+
+  auto dimSizes = mhlo::getDimSizesOfTensor(rewriter, op, adaptor.self());
+  auto loc = op.getLoc();
+  auto one =
+      rewriter.create<arith::ConstantOp>(loc, rewriter.getI32IntegerAttr(1));
+  Value numel = one;
+  for (auto& d : dimSizes) {
+    numel = rewriter.create<arith::MulIOp>(loc, numel, d);
+  }
+  rewriter.replaceOpWithNewOp<arith::ExtSIOp>(
```

Comment:
can skip this extension if the same element type?

Reply:
why is this needed...

Reply:
DimSizes have i32 element type while the op returns i64. That is why it's needed.